libminifi/src/io/win/ClientSocket.cpp

```diff
@@ -242,11 +237,17 @@ int16_t Socket::initialize() {
   hints.ai_protocol = 0; /* any protocol */
 
   int errcode = getaddrinfo(requested_hostname_.c_str(), 0, &hints, &addr_info_);
```

Comment:
The lifetime of `addr_info_` was changed from "from initialization to either the first failing `connect` or the end of `this`" to "during initialization".

Reply:
That's great, but in that case I think it just needs to be a local variable and not a member.

Reply:
The value is accessed by `createConnection` as a data member and I didn't want to risk breaking anything during this bugfix by changing the signature to take an additional argument.

I'd prefer to later rewrite the whole class, that will most likely eliminate the need to pass the result of `getaddrinfo` to the function.

I can do additional investigation about the required effort if you insist on this change.

Reply:
done in 6512bec. Please excuse my confusion while writing the previous comment